src/webgpu/shader/execution/memory_model/memory_model_setup.ts

```diff
@@ -191,19 +202,28 @@ export class MemoryModelTester {
     this.test = t;
     this.params = params;
 
-    testShader = shaderPreamble(accessValueType) + testShader;
-    resultShader = shaderPreamble(accessValueType) + resultShader;
+    const workgroupXSize = Math.min(params.workgroupSize, t.device.limits.maxComputeWorkgroupSizeX);
```

Comment:
Interestingly, the OpenCL CTS also recently had to be updated to accommodate devices that only handle up to 128-size workgroups.

See https://github.com/KhronosGroup/OpenCL-CTS/pull/1846